packages/grid/x-data-grid/src/tests/cells.DataGrid.test.tsx

```diff
@@ -184,6 +185,42 @@ describe('<DataGrid /> - Cells', () => {
     }).toWarnDev(['MUI: The cell with id=1 and field=brand received focus.']);
   });
 
+  it('selected cell  should alwys been in the dom,', function test() {
+    if (isJSDOM) {
+      this.skip();
+    }
+    const rowHeight = 50;
+    const defaultData = getBasicGridData(20, 20);
+
+    render(
+      <div style={{ width: 300, height: 300 }}>
+        <DataGrid columns={defaultData.columns} rows={defaultData.rows} rowHeight={rowHeight} />
+      </div>,
+    );
+
+    const virtualScroller = document.querySelector('.MuiDataGrid-virtualScroller')!;
+
+    const thirdRowCell = getCell(1, 3);
+    userEvent.mousePress(thirdRowCell);
+
+    const activeElementTextContent = document.activeElement?.textContent;
+    const columnWidth = document.activeElement!.clientWidth;
+
+    // check for row virtualazition
+    const scrollTop = 10 * rowHeight;
+    virtualScroller.scrollTop = scrollTop;
+    virtualScroller.dispatchEvent(new Event('scroll'));
```

Comment:
Any way to use `fireEvent` like the rest of the test code does?

https://stackoverflow.com/a/56233748/6303229

Also applies below